consensus/consensus_v2.go

```diff
@@ -679,13 +680,66 @@ func (consensus *Consensus) commitBlock(blk *types.Block, committedMsg *FBFTMess
 	return nil
 }
 
+func (consensus *Consensus) rotateLeader(epoch *big.Int) {
+	prev := consensus.GetLeaderPubKey()
+	curNumber := consensus.Blockchain.CurrentHeader().Number().Uint64()
+	utils.Logger().Info().Msgf("[Rotating leader] epoch: %v rotation:%v numblocks:%d", epoch.Uint64(), consensus.Blockchain.Config().IsLeaderRotation(epoch), consensus.Blockchain.Config().LeaderRotationBlocksCount)
+	if consensus.Blockchain.Config().IsLeaderRotation(epoch) {
```

Comment:
Why are we checking this again?

Reply:
This check is cheap, we can keep it. Previously this function was called from different places, so we needed it.